src/file.ts

```diff
@@ -3560,15 +3564,31 @@ class File extends ServiceObject<File> {
       typeof optionsOrCallback === 'function' ? optionsOrCallback : callback;
     const options =
       typeof optionsOrCallback === 'object' ? optionsOrCallback : {};
-
-    const writable = this.createWriteStream(options)
-      .on('error', callback!)
-      .on('finish', callback!);
-    if (options.onUploadProgress) {
-      writable.on('progress', options.onUploadProgress);
+    const isMultipart = options.resumable === false;
+    const returnValue = retry(async (bail: (err: Error) => void) => {
+      await new Promise<void>((resolve, reject) => {
+        const writable = this.createWriteStream(options)
```

Comment:
Do end users ever use streams directly, or is it more typical to use save?

Reply:
That's the open question raised in today's meeting. We aren't sure. It might not a use-case for `file.save()` and @shaffeeullah is looking into it.

Reply:
PR here: https://github.com/googleapis/nodejs-storage/pull/1388